Lib/lzma.py

```diff
@@ -131,6 +131,11 @@ def __init__(self, filename=None, mode="r", *,
                 trailing_error=LZMAError, format=format, filters=filters)
             self._buffer = io.BufferedReader(raw)
 
+        if not isinstance(filename, (str, bytes)):
```

Comment:
What if it is a path-like object?

Also, if it is a file, we perhaps can use its name. Actually, we perhaps can simply use `self._fp.name`.